### PR TITLE
[SPARK-15464][ML][MLlib][SQL][Tests] Replace SQLContext and SparkContext with SparkSession using builder pattern in python test code

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

Replace SQLContext and SparkContext with SparkSession using builder pattern in python test code.
## How was this patch tested?

Existing test.

### CODE REVIEW

**Comment by rxin** [comment]
cc @andrewor14

File: python/pyspark/ml/clustering.py [hunk 1]
```diff
@@ -933,21 +933,20 @@ def getKeepLastCheckpoint(self):
...
     import pyspark.ml.clustering
-    from pyspark.context import SparkContext
-    from pyspark.sql import SQLContext
+    from pyspark.sql import SparkSession
     globs = pyspark.ml.clustering.__dict__.copy()
     # The small batch size here ensures that we see multiple batches,
     # even in these small test examples:
-    sc = SparkContext("local[2]", "ml.clustering tests")
-    sqlContext = SQLContext(sc)
+    spark = SparkSession.builder.master("local[2]").appName("ml.clustering tests").getOrCreate()
```
Comment (techaddict) [inline python/pyspark/ml/clustering.py]: Can you make this multi line

Comment (WeichenXu123) [inline python/pyspark/ml/clustering.py]: @techaddict Done.

**Comment by andrewor14** [comment]
Looks good. Merging into master 2.0.

**Pull request closed without merging**